crates/re_viewport/src/context_menu/actions.rs

```diff
@@ -375,14 +377,147 @@ impl ContextMenuAction for MoveContentsToNewContainerAction {
     }
 }
 
-impl MoveContentsToNewContainerAction {
-    fn target_container_id_and_position(
-        ctx: &ContextMenuContext<'_>,
-    ) -> Option<(ContainerId, usize)> {
-        ctx.clicked_item
-            .clone()
-            .try_into()
-            .ok()
-            .and_then(|c| ctx.viewport_blueprint.find_parent_and_position_index(&c))
+// ---
+
+/// Create a new space view containing the selected entities.
+///
+/// The space view is created next to the clicked item's parent view (if a data result was clicked).
+pub(super) struct AddEntitiesToNewSpaceViewAction;
+
+impl ContextMenuAction for AddEntitiesToNewSpaceViewAction {
+    fn supports_multi_selection(&self, _ctx: &ContextMenuContext<'_>) -> bool {
+        true
+    }
+
+    fn supports_item(&self, _ctx: &ContextMenuContext<'_>, item: &Item) -> bool {
+        matches!(item, Item::DataResult(_, _) | Item::InstancePath(_))
     }
+
+    fn ui(&self, ctx: &ContextMenuContext<'_>, ui: &mut Ui) -> Response {
+        let space_view_class_registry = ctx.viewer_context.space_view_class_registry;
+
+        let recommended_space_view_classes = recommended_space_views_for_selection(ctx);
+        let other_space_view_classes: IntSet<_> = space_view_class_registry
+            .iter_registry()
+            .map(|entry| entry.class.identifier())
+            .collect::<IntSet<SpaceViewClassIdentifier>>()
+            .difference(&recommended_space_view_classes)
+            .cloned()
+            .collect();
+
+        ui.menu_button("Add to new space view", |ui| {
+            let buttons_for_space_view_classes =
+                |ui: &mut egui::Ui, space_view_classes: &IntSet<SpaceViewClassIdentifier>| {
+                    for (identifier, display_name) in space_view_classes
+                        .iter()
+                        .map(|identifier| {
+                            (
+                                identifier,
+                                space_view_class_registry
+                                    .get_class_or_log_error(identifier)
+                                    .display_name(),
+                            )
+                        })
+                        .sorted_by_key(|(_, display_name)| display_name.to_owned())
+                    {
+                        if ui.button(display_name).clicked() {
+                            create_space_view_with_entities(ctx, *identifier);
+                        }
+                    }
+                };
+
+            ui.label(egui::WidgetText::from("Recommended:").italics());
+            if recommended_space_view_classes.is_empty() {
+                ui.label("None");
+            } else {
+                buttons_for_space_view_classes(ui, &recommended_space_view_classes);
+            }
+
+            ui.label(egui::WidgetText::from("Others:").italics());
```

Comment:
should display this only if there's any element in `other_space_view_classes`

Reply:
I figured an universally displayable entity was not possible, but yeah better safe than sorry.